src/devices/src/legacy/serial.rs

```diff
@@ -244,10 +240,26 @@ impl Serial {
         }
     }
 
+    fn recv_bytes(&mut self) -> io::Result<usize> {
+        if let Some(input) = self.input.as_mut() {
+            let mut out = [0u8; 32];
+            return match input.read(&mut out) {
```

Comment:
This could be run in a loop to minimize epoll events when there is a lot of data pending on the pipe

Reply:
> However, I can not tell if we have any data about the need to optimize it and I am not sure if that would help significantly for any serial console use case, at the moment.

It's just a simple optimizing suggestion from looking at the code.

> Furthermore, if you think the serial input needs optimizations on the processing path, I think we have to address this in a separate issue/PR.

Sure, if you feel this is complicated, we can do it in a separate PR.